packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts

```diff
@@ -24,3 +30,37 @@ export const createSceneRootEntity = (): Promise<CreateEntityCommandOutput> | un
 
   return getGlobalSettings().twinMakerSceneMetadataModule!.createSceneEntity(input);
 };
+
+export const checkIfEntityAvailable = async (
+  entityId: string,
+  sceneMetadataModule: TwinMakerSceneMetadataModule,
+): Promise<boolean> => {
+  try {
+    await sceneMetadataModule.getSceneEntity({ entityId });
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+export const prepareWorkspace = async (sceneMetadataModule: TwinMakerSceneMetadataModule): Promise<void> => {
+  // Check if root entities exist
+  const checkRoots = await Promise.all([
+    checkIfEntityAvailable(SCENE_ROOT_ENTITY_ID, sceneMetadataModule),
+    checkIfEntityAvailable(LAYER_ROOT_ENTITY_ID, sceneMetadataModule),
+  ]);
+
+  // Create root entities that do not exist
+  const createRootRequests: Promise<CreateEntityCommandOutput>[] = [];
+  if (!checkRoots[0]) {
+    createRootRequests.push(
+      sceneMetadataModule.createSceneEntity({ entityId: SCENE_ROOT_ENTITY_ID, entityName: SCENE_ROOT_ENTITY_NAME }),
+    );
+  }
+  if (!checkRoots[1]) {
+    createRootRequests.push(
+      sceneMetadataModule.createSceneEntity({ entityId: LAYER_ROOT_ENTITY_ID, entityName: LAYER_ROOT_ENTITY_NAME }),
+    );
+  }
+  await Promise.all(createRootRequests);
```

Comment:
We a re assuming here that all promises will be resolved successfully. However, if any of the promises in the createRootRequests array reject, it will throw an unhandled promise rejection error. Either you can add try catch here or in the caller method.

Reply:
the caller should catch. the current component doesn't support showing error state. I'll do it in a separate PR